patches/server/0629-Remove-streams-from-SensorNearest.patch

```diff
@@ -63,6 +63,35 @@ index 66c90013e52170a657b1a5dbdb99748a19fe55e8..ffd83db0a419ab589e89feeddd3fb038
      }
  
      @Override
+diff --git a/src/main/java/net/minecraft/world/entity/ai/sensing/NearestVisibleLivingEntitySensor.java b/src/main/java/net/minecraft/world/entity/ai/sensing/NearestVisibleLivingEntitySensor.java
+index 52bf24959de40fe52a721b8ffa3124cc8db63a0d..350e46596711b87c44a40c4be47f4a7c5df76f46 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/sensing/NearestVisibleLivingEntitySensor.java
++++ b/src/main/java/net/minecraft/world/entity/ai/sensing/NearestVisibleLivingEntitySensor.java
+@@ -25,11 +25,19 @@ public abstract class NearestVisibleLivingEntitySensor extends Sensor<LivingEnti
+     }
+ 
+     private Optional<LivingEntity> getNearestEntity(LivingEntity entity) {
+-        return this.getVisibleEntities(entity).flatMap((list) -> {
+-            return list.stream().filter((livingEntity2) -> {
+-                return this.isMatchingEntity(entity, livingEntity2);
+-            }).min(Comparator.comparingDouble(entity::distanceToSqr));
+-        });
++        Optional<List<LivingEntity>> visibleEntities = this.getVisibleEntities(entity); // Paper Start - Remove Streams
++        if (visibleEntities.isEmpty()) {
++            return Optional.empty();
++        } else {
++            List<LivingEntity> visibleEntityList = visibleEntities.get();
++            visibleEntityList.removeIf((visibleEntity) -> !this.isMatchingEntity(entity, visibleEntity));
++            if (visibleEntityList.isEmpty()) {
++                return Optional.empty();
++            }
++
++            visibleEntityList.sort(Comparator.comparingDouble(entity::distanceToSqr));
++            return Optional.of(visibleEntityList.get(0));
```

Comment:
```suggestion
+            return Optional.of(Collections.min(visibleEntityList, Comparator.comparingDouble(entity::distanceToSqr)));
```

Reply:
Thank you, that is perfect! Unsure what I was thinking over here.